### PR TITLE
787 - Enhancements for the User Task Runs

### PR DESCRIPTION
**Pull request by diablouma**
This PR includes:
 - Who is it currently assigned to? (user_group and/or user_id)_
 - The "Audit Log" which is the events field of the UserTaskRun
 - Assignment history and/or cancellation history
 - If it has a specific EXCEPTION that it throws upon cancellation
 - The results after it is completed
 - When the UserTaskRun was created (i.e. arrival time)
 - When the UserTaskRun was completed or cancelled (if it isn't still pending)

### CODE REVIEW

File: dashboard/src/app/(authenticated)/(diagram)/components/Modals/UserTaskRun.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import { useModal } from '@/app/(authenticated)/(diagram)/hooks/useModal'
+import { Dialog, DialogPanel, DialogTitle } from '@headlessui/react'
+import { DialogBody } from 'next/dist/client/components/react-dev-overlay/internal/components/Dialog'
+import { XMarkIcon } from '@heroicons/react/24/outline'
+import { getVariable, getVariableValue } from '@/app/utils'
+import { UserTaskRunStatus } from 'littlehorse-client/dist/proto/user_tasks'
+
+export const UserTaskRun: FC<UserTaskModal> = ({ data, nodeRun, userTaskNode }) => {
+  const { showModal, setShowModal } = useModal()
+  const refDiv = useRef(null)
```
Comment (mijailr) [inline dashboard/src/app/(authenticated)/(diagram)/components/Modals/UserTaskRun.tsx:12]: I wonder why is this necessary if TaskRun doesn't need it?

File: dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTask.tsx [hunk 2]
```diff
@@ -26,11 +29,10 @@ const Node: FC<NodeProps> = ({ data, selected }) => {
             </Link>
           </div>
           <div className="mb-2 flex gap-2 text-nowrap">
-            {userTask.userGroup && (
-              <div className="flex items-center justify-center">Group: {getVariable(userTask.userGroup)}</div>
-            )}
-            {userTask.userId && (
-              <div className="flex items-center justify-center">User: {getVariable(userTask.userId)}</div>
+            {nodeRun ? (
+              <UserTaskRunDetails userTaskNode={userTask} nodeRun={nodeRun} />
```
Comment (mijailr) [inline dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTask.tsx:33]: This component should handle `nodeRun` with `undefined` value. Please handle the case on a parent component to simplify this one.

File: dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTaskDefDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { getVariable } from '@/app/utils'
+
+export const UserTaskDefDetails: FC<{ userTask: UserTaskNode }> = ({ userTask }) => {
+  return (
+    <>
+      {userTask.userGroup && (
+        <div className="flex items-center justify-center">Group: {getVariable(userTask.userGroup)}</div>
+      )}
+
+      {userTask.userId && <div className="flex items-center justify-center">User: {getVariable(userTask.userId)}</div>}
```
Comment (mijailr) [inline dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTaskDefDetails.tsx:12]: This content is similar to `UserTaskRunDetails` can it be a shared component instead?

**Review by mijailr — APPROVED** [review]
Looks good, just some optional comments.

**Comment by coltmcnealy-lh** [comment]
We could use a new ticket that differentiates what we show on the `WfSpec` from the `WfRun`

File: dashboard/src/app/(authenticated)/(diagram)/components/Modals/UserTaskRun.tsx (hunk 1 above)
Comment (diablouma) [reply dashboard/src/app/(authenticated)/(diagram)/components/Modals/UserTaskRun.tsx:12]: I was needing certain properties like nodeRun.endTime and the cancellationException which are not part of the UserTask object itself.

File: dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTaskDefDetails.tsx (hunk 3 above)
Comment (diablouma) [reply dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTaskDefDetails.tsx]: extracted :)

File: dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTask.tsx (hunk 2 above)
Comment (diablouma) [reply dashboard/src/app/(authenticated)/(diagram)/components/NodeTypes/UserTask/UserTask.tsx:33]: Mijail are you saying that I should have a component that receives the nodeRun and handle both cases internally instead of handling them at the UserTask.tsx?
